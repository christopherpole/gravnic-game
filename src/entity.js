const NONE = 0;
const FLOOR = 1;
const BLOCK = 2;

module.exports = {
  NONE,
  FLOOR,
  BLOCK,
};
