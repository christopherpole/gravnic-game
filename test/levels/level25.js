import { MOVE_RIGHT, MOVE_DOWN, ENTITIES, levelIsComplete, makeMoves } from '../../src/index';

describe('Level 25', () => {
  it('Matches the correct snapshot', () => {
    const gameState = [
      [
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 2 },
          movableEntity: { entityId: ENTITIES.BLOCK.id, color: '#800080', id: 1 },
        },
        { staticEntity: { entityId: ENTITIES.FLOOR.id, id: 3 }, movableEntity: null },
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 5 },
          movableEntity: { entityId: ENTITIES.CRATE.id, id: 4 },
        },
        { staticEntity: { entityId: ENTITIES.FLOOR.id, id: 6 }, movableEntity: null },
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 8 },
          movableEntity: { entityId: ENTITIES.CRUSHER.id, id: 7 },
        },
      ],
      [
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 10 },
          movableEntity: { entityId: ENTITIES.CRUSHER.id, id: 9 },
        },
        { staticEntity: { entityId: ENTITIES.FLOOR.id, id: 11 }, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.FLOOR.id, id: 12 }, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.BLACK_HOLE.id, id: 13 }, movableEntity: null },
      ],
      [
        { staticEntity: null, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.FLOOR.id, id: 14 }, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.FLOOR.id, id: 15 }, movableEntity: null },
        { staticEntity: { entityId: ENTITIES.FLOOR.id, id: 16 }, movableEntity: null },
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 18 },
          movableEntity: { entityId: ENTITIES.BLOCK.id, color: '#0000FF', id: 17 },
        },
      ],
      [
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
        {
          staticEntity: { entityId: ENTITIES.FLOOR.id, id: 20 },
          movableEntity: { entityId: ENTITIES.CRUSHER.id, id: 19 },
        },
        { staticEntity: null, movableEntity: null },
        { staticEntity: null, movableEntity: null },
      ],
    ];

    const gameStates = makeMoves(gameState, [MOVE_RIGHT, MOVE_DOWN]);
    expect(gameStates).toMatchSnapshot();
    expect(
      levelIsComplete(
        gameStates[gameStates.length - 1][gameStates[gameStates.length - 1].length - 1],
      ),
    ).toBe(true);
  });
});
