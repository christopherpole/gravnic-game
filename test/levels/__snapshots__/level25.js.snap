// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level 25 Matches the correct snapshot 1`] = `
Array [
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRATE",
            "id": 4,
            "moved": true,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 13,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 19,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 20,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRATE",
            "id": 4,
            "moved": true,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 9,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 13,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 19,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 20,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRATE",
            "id": 4,
            "moved": true,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 9,
            "shrinking": true,
          },
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 13,
            "shrinking": true,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 19,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 20,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRATE",
            "id": 4,
            "moved": true,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 13,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 19,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 20,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRATE",
            "id": 4,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 13,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 19,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 20,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRATE",
            "id": 4,
            "moved": true,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 13,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#0000FF",
            "entityId": "BLOCK",
            "id": 17,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 19,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 20,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRATE",
            "id": 4,
            "moved": true,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 13,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": Object {
            "color": "#800080",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 19,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 20,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRATE",
            "id": 4,
            "moved": true,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 13,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 19,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 20,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 6,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 8,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 10,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 11,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRATE",
            "id": 4,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 12,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 13,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 14,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 15,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 16,
          },
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 7,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 18,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": Object {
            "entityId": "CRUSHER",
            "id": 19,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 20,
          },
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
        Object {
          "movableEntity": null,
          "staticEntity": null,
        },
      ],
    ],
  ],
]
`;
